File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3005_count-elements-with-maximum-frequency/solutions/c/solution.c
#include <stdio.h>

int maxFrequencyElements(int* nums, int numsSize) {
    int count[101] = {0};
    int maxFreq = 0;
    
    for (int i = 0; i < numsSize; i++) {
        count[nums[i]]++;
        if (count[nums[i]] > maxFreq) {
            maxFreq = count[nums[i]];
        }
    }
    
    int result = 0;
    for (int i = 1; i <= 100; i++) {
        if (count[i] == maxFreq) {
            result += count[i];
        }
    }
    
    return result;
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3005_count-elements-with-maximum-frequency/solutions/java/solution.java
import java.util.HashMap;

class Solution {
    public int maxFrequencyElements(int[] nums) {
        HashMap<Integer, Integer> freqMap = new HashMap<>();
        int maxFreq = 0;
        
        // Count the frequency of each number
        for (int num : nums) {
            int count = freqMap.getOrDefault(num, 0) + 1;
            freqMap.put(num, count);
            maxFreq = Math.max(maxFreq, count);
        }
        
        int maxFreqCount = 0;
        
        // Calculate the total count of numbers with the maximum frequency
        for (int count : freqMap.values()) {
            if (count == maxFreq) {
                maxFreqCount += count;
            }
        }
        
        return maxFreqCount;
    }
}
